### PR TITLE
CODENVY-478 remove che-core-api-account

### PR DESCRIPTION
**Pull request by skabashnyuk**

### CODE REVIEW

**Comment by benoitf** [comment]
probably should include removal of https://github.com/codenvy/codenvy/blob/master/dashboard/src/components/api/codenvy-account.factory.js and usage

**Comment by benoitf** [comment]
+1 for me

File: assembly/onpremises-ide-packaging-war-factory/src/main/java/com/codenvy/onpremises/factory/FactoryServlet.java [hunk 1]
```diff
@@ -71,11 +71,10 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
         }
 
         // DO NOT REMOVE! This log will be used in statistic analyzing
-        LOG.info("EVENT#factory-accepted# WS#{}# REFERRER#{}# FACTORY#{}# ORG-ID#{}# AFFILIATE-ID#{}# WS-LOCATION#{}# WS-TYPE#{}#",
+        LOG.info("EVENT#factory-accepted# WS#{}# REFERRER#{}# FACTORY#{}# # AFFILIATE-ID#{}# WS-LOCATION#{}# WS-TYPE#{}#",
```
Comment (sleshchenko) [inline assembly/onpremises-ide-packaging-war-factory/src/main/java/com/codenvy/onpremises/factory/FactoryServlet.java]: As far as I know we don't have analitics for 4.x version. Maybe we can just remove this logging with outdated AFFILIATE-ID and WS-TYPE

File: assembly/onpremises-ide-packaging-war-platform-api/src/main/java/com/codenvy/api/deploy/OnPremisesIdeApiServletModule.java [hunk 2]
```diff
@@ -48,14 +46,6 @@ protected void configureServlets() {
                 .through(com.codenvy.service.http.WorkspaceIdEnvironmentInitializationFilter.class);
         filterRegex("^/workspace/(?!config$|runtime$|.*:.*$).+")
                 .through(com.codenvy.service.http.WorkspaceIdEnvironmentInitializationFilter.class);
-        filterRegex("^/(account|creditcard)/(?!find|list).+").through(new AccountIdEnvironmentInitializationFilter(),
```
Comment (sleshchenko) [inline assembly/onpremises-ide-packaging-war-platform-api/src/main/java/com/codenvy/api/deploy/OnPremisesIdeApiServletModule.java]: also remove please account from mapping for LoginFilter here https://github.com/codenvy/codenvy/pull/181/files#diff-d7ca62cd2a04167dc47ecffc7497eb03R61 and maybe billing, creditcard and other like that

File: core/codenvy-hosted-api-admin/src/main/java/com/codenvy/api/dao/ldap/AdminUserDaoImpl.java [hunk 3]
```diff
@@ -54,7 +53,7 @@
     private final UserLdapPagination userLdapPagination;
 
     @Inject
-    public AdminUserDaoImpl(AccountDao accountDao,
+    public AdminUserDaoImpl(
```
Comment (sleshchenko) [inline core/codenvy-hosted-api-admin/src/main/java/com/codenvy/api/dao/ldap/AdminUserDaoImpl.java]: Fix formatting here

File: core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/dao/mongo/AccountDaoImpl.java [hunk 4]
```diff
@@ -1,408 +0,0 @@
...
- *
- * @author Max Shaposhnik
- * @author Eugene Voevodin
- * @author Alexander Garagatyi
- */
-@Singleton
-public class AccountDaoImpl implements AccountDao {
-
-    private static final Logger LOG                = LoggerFactory.getLogger(AccountDaoImpl.class);
-    private static final String ACCOUNT_COLLECTION = "organization.storage.db.account.collection";
```
Comment (sleshchenko) [inline core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/dao/mongo/AccountDaoImpl.java]: Did you prepare pull request for deployment project? 
Should we remove mongo collection for accounts?

File: assembly/onpremises-ide-packaging-war-factory/src/main/java/com/codenvy/onpremises/factory/FactoryServlet.java (hunk 1 above)
Comment (benoitf) [inline assembly/onpremises-ide-packaging-war-factory/src/main/java/com/codenvy/onpremises/factory/FactoryServlet.java]: I would keep a log entry. This is for now the only way to monitor factories that are being invoked by users and to know from where it comes

File: core/codenvy-hosted-sso-client/src/main/java/com/codenvy/auth/sso/client/EnvironmentContextResolver.java [hunk 5]
```diff
@@ -27,6 +27,6 @@
     @Override
     public RolesContext getRequestContext(HttpServletRequest request) {
         EnvironmentContext context = EnvironmentContext.getCurrent();
-        return new RolesContext(context.getWorkspaceId(), context.getAccountId());
+        return new RolesContext(context.getWorkspaceId(), null);
```
Comment (sleshchenko) [inline core/codenvy-hosted-sso-client/src/main/java/com/codenvy/auth/sso/client/EnvironmentContextResolver.java]: Do you plan to remove account id from RolesContext?

**Comment by sleshchenko** [comment]
other OK for me

File: core/codenvy-hosted-sso-client/src/main/java/com/codenvy/auth/sso/client/EnvironmentContextResolver.java (hunk 5 above)
Comment (skabashnyuk) [inline core/codenvy-hosted-sso-client/src/main/java/com/codenvy/auth/sso/client/EnvironmentContextResolver.java]: I'm planning to remove RolesContext at all.

**Comment by vparfonov** [comment]
ok

**Pull request merged**